docs/src/modules/spring/pages/views.adoc

```diff
@@ -1 +1,472 @@
+
+Views can be defined from any of the following:
+
+* xref:value-entity[Value Entities state changes]
+* xref:event-sourced-entity[Event Sourced Entity events]
+* xref:topic-view[Messages received from subscribing to topics on a broker]
+
+The remainder of this page describes:
+
+* <<transform-results>>
+* <<#changing>>
+* <<#query>>
+
+IMPORTANT: Be aware that Views are not updated immediately when Entity state changes. Kalix does update Views as quickly as possible, but it is not instant and can take up to a few seconds for the changes to become visible in the query results. View updates might also take more time during failure scenarios than during normal operation.
+
+[#value-entity]
+== Creating a View from a Value Entity
+
+Consider an example of a Customer Registry service with a `customer` Value Entity. When `customer` state changes, the entire state is emitted as a value change. Value changes update any associated Views. To create a View that lists customers by their name:
+
+* <<_define_the_view>> for a service that selects customers by name and associates a table name with the View. The table is created and used by Kalix to store the View.
+
+
+This example assumes the following `customer` exists
+
+[source,java,indent=0]
+.src/main/java/customer/api/Customer.java
+----
+include::example$spring-customer-registry-views-quickstart/src/main/java/customer/api/Customer.java[tags=class]
+----
+
+As well as a Value Entity service `CustomerEntity.java` that will produce the state changes consumed by the View:
+
+
+=== Define the View
+
+To get a View of multiple customers by their name, you need the set the following:
+
+ - Define the name of the table with `@Table`.
+ - Subscribe to the Value Entity with `@Subscribe.ValueEntity`.
+ - Extend from the class `View`.
+ - Define an endpoint for the view with `@GetMapping`. Is this optional? Can the view be called by a component?
+ - Define the query with `@Query`. 
+
+.src/main/java/customer/view/CustomerByNameView.java
+[source,java]
+----
+include::example$spring-customer-registry-views-quickstart/src/main/java/customer/view/CustomerByNameView.java[tags=class]
+----
+<1> defining table name
+<2> subscribing to `CustomerEntity`
+<3> extending from `View`
+<4> defining endpoint
+<5> defining the query
+<6> note that no return is needed
+
+IMPORTANT: Note that the return value of the method is `null`. You may ask yourself,  how is it that the endpoint respond with any `Customer` at all? The fact is that this customer object is created in the proxy and not in the View. When you call this endpoint, the view calls the proxy, which in turn calls the database. When the proxy receives the response, it sends it directly to you without any further intervention from the View. Therefore, null is valid as a return value in the endpoint of a View. The choice of null is a way to signal that the response doesn't come from the return of this method. 
+
+TODO: ? how do we know there is only one Customer? 
```

Comment:
We don't. That's specific to the application. It will only return one customer if there is a unique match, typically when search on a unique key. 

Actually, not a very realistic use case on a system that only support one unique key (only EntityKey is guaranteed to be unique).

I think we should return a `Flux<Customer>`. That's a more sensible default of a query.

Note that the gRPC version is returning a `stream CustomerState`. So here we should return a `Flux<Cutomer>`